cmd/scaffold.go

```diff
@@ -203,8 +198,7 @@ func (fnb *FlowNodeBuilder) EnqueueNetworkInit(ctx context.Context) {
 
 		fnb.Network = net
 
-		idRefresher := p2p.NewNodeIDRefresher(fnb.Logger, fnb.State, net.SetIDs)
-		idEvents := gadgets.NewIdentityDeltas(idRefresher.OnIdentityTableChanged)
+		idEvents := gadgets.NewIdentityDeltas(fnb.Middleware.UpdateAllowList)
```

Comment:
so originally, the gadget used to update the `network` with the new ids and the `network` would then update `middleware`. 
Why are we directly updating `middleware` here, wouldn't the `overlay` be out of sync at that point with the change.

Instead, if the `IdentifierProvider` provided a way to `update` its list, then both middleware and network could use it.